addons/website/models/website_page.py

```diff
@@ -284,19 +284,29 @@ def _search_get_detail(self, website, order, options):
     @api.model
     def _search_fetch(self, search_detail, search, limit, order):
         with_description = 'description' in search_detail['mapping']
-        results, count = super()._search_fetch(search_detail, search, limit, order)
+        # Cannot rely on the super's _search_fetch because the search must be
+        # performed among the most specific pages only.
+        fields = search_detail['search_fields']
+        base_domain = search_detail['base_domain']
+        domain = self._search_build_domain(base_domain, search, fields, search_detail.get('search_extra'))
+        # TODO In 16.0 do not rely on _filter_duplicate_pages.
+        most_specific_pages = self.env['website'].with_context(_filter_duplicate_pages='url' not in order)._get_website_pages(
+            domain=expression.AND(base_domain), order=order
+        )
+        results = most_specific_pages.sudo().filtered_domain(domain)
```

Comment:
Could you explain to me why the sudo is needed here? If I'm not wrong, prior to this commit, the sudo was only performed if `requires_sudo` was in the search detail. Why not here?

Reply:
Ultimately removed that `sudo` because the result of `_get_website_pages` is already `sudo`.